dll/win32/iernonce/registry.cpp

```diff
@@ -37,21 +38,43 @@ BOOL RunOnceExEntry::Delete(
 
 BOOL RunOnceExEntry::Exec() const
 {
-    // CreateProcessW requires lpCommandLine is non-const, so copy one.
-    CStringW ValueCopy = m_Value;
-    STARTUPINFOW si = { sizeof(si) };
-    PROCESS_INFORMATION pi = { 0 };
-    BOOL bSuccess = CreateProcessW(NULL, ValueCopy.GetBuffer(), NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi);
-    ValueCopy.ReleaseBuffer();
+    CStringW CommandLine;
+    if (wcsncmp(m_Value, L"||", 2) == 0)
+    {
+        // remove the prefix;
+        CommandLine = (LPCWSTR)m_Value + 2;
+    }
+    else
+    {
+        CommandLine = m_Value;
+    }
+
+    // FIXME: SHEvaluateSystemCommandTemplate is not implemented
+    //        using PathGetArgsW, PathRemoveArgsW as a workaround.
+    LPWSTR szCommandLine = CommandLine.GetBuffer();
+    LPCWSTR szParam = PathGetArgsW(szCommandLine);
+    PathRemoveArgsW(szCommandLine);
+
+    SHELLEXECUTEINFOW Info = { 0 };
+    Info.cbSize = sizeof(Info);
+    Info.fMask = SEE_MASK_NOCLOSEPROCESS;
+    Info.lpFile = szCommandLine;
+    Info.lpParameters = szParam;
+    Info.nShow = SW_SHOWNORMAL;
+
+    BOOL bSuccess = ShellExecuteExW(&Info);
+
+    CommandLine.ReleaseBuffer();
+
     if (!bSuccess)
     {
         return FALSE;
     }
-    CloseHandle(pi.hThread);
+    //CloseHandle(pi.hThread);
```

Comment:
I think you can remove that one altogether, since ShellExecuteEx does that internally.